dsl_parser/rel_graph.py

```diff
@@ -823,9 +828,13 @@ def _verify_and_get_connection_type(relationship):
     return connection_type
 
 
-def _relationship_type_hierarchy_includes_one_of(relationship, expected_types):
-    relationship_type_hierarchy = relationship['type_hierarchy']
-    return any([relationship_type in expected_types
+def _relationship_type_hierarchy_includes_one_of(relationship,
+                                                 expected_types):
+    def partial_match_in_list(source, item_list):
```

Comment:
Not 100% what's going on here. Is this just a 2 layer list search? Can probably use a comment (or better yet, a refactor)

Reply:
Commented

Reply:
improved the naming and added a clarification